src/common/Utilities/StringConvert.h

```diff
@@ -0,0 +1,178 @@
+/*
+ * This file is part of the TrinityCore Project. See AUTHORS file for Copyright information
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef TRINITY_STRINGCONVERT_H
+#define TRINITY_STRINGCONVERT_H
+
+#include "Define.h"
+#include "Errors.h"
+#include "Optional.h"
+#include "Util.h"
+#include <charconv>
+#include <string>
+#include <string_view>
+#include <type_traits>
+
+namespace Trinity::Impl::StringConvertImpl
+{
+    template <typename T, typename = void> struct For
+    {
+        static_assert("Unsupported type used for ToString or StringTo");
```

Comment:
This becomes static_assert(true)

Reply:
static_assert(!std::is_same_v<T, T>, "bla");

Reply:
consider adding

```cpp
template <typename T>
struct dependant_false { constexpr static const bool value = false; };

template <typename T>
constexpr static const bool dependant_false_v = dependant_false<T>::value;
```

to advstd?

Reply:
is that actually in c++20 @Warpten?

Reply:
and yeah i brainfarted and forgot the is_same_v

Reply:
Nah it's not, but it's so convenient we might as well. Or add another header, at least the intent is clearer that way